src/scenes/Root/Header/HeaderSearch/HeaderSearch.tsx

```diff
@@ -15,13 +16,30 @@ const useStyles = makeStyles(({ palette, spacing }) => ({
 
 export const HeaderSearch: FC = () => {
   const classes = useStyles();
+  const navigate = useNavigate();
+  const location = useLocation();
+  const locationState: { searchValue?: string } | null = location.state;
+
+  const goToSearchPage = (value: string) =>
+    navigate(
+      {
+        pathname: 'search',
+      },
+      {
+        state: {
+          searchValue: value,
+        },
+      }
+    );
```

Comment:
> I had to store the search input value somewhere so I put it in the router state. I don't have a strong preference for where this is stored. Initially I tried to put it as a query param but then had to deal with syncing this with the search input so I switched it to router state. If you know a different way to do this that is cleaner I can change this.


Yeah you can use the `makeQueryHandler` function which produces a "setState"-like hook

```tsx
// outside of component
const useSearch = makeQueryHandler({
  search: StringParam,
});

// in component
const [val, setVal] = useSearch();
```
This hook can be used in multiple places too.

Reply:
Here's an example in code
https://github.com/SeedCompany/cord-field/blob/a87f0ece1aef8bc5a070ca6dfa41d440f3a5260d/src/scenes/Projects/List/ProjectFilterOptions.tsx#L8-L12

Reply:
Hi, I've had difficulty with the useSearchParams functions from "react-router-dom" inside the makeQueryHandler hook. It keeps resetting the url to "/" instead of appending the url parameters to the current url. Is there a parameter I need to add to navigate options to not reset the url? I've tried to set replace to false but I think this is for url params and not the url.

Reply:
I was able to reproduce that @jmc637, sorry for the delay. This is a problem with react-router. I applied a workaround in our makeQueryHandler code.